src/gt4py/frontend/gtscript_frontend.py

```diff
@@ -470,8 +474,7 @@ def visit_Call(self, node: ast.Call, *, target_node=None):
                 "Function calls are not supported in arguments to function calls",
                 loc=gt_ir.Location.from_ast_node(node),
             )
-
-        elif call_name not in self.context and not hasattr(self.context[call_name], "_gtscript_"):
+        elif call_name not in self.context or not hasattr(self.context[call_name], "_gtscript_"):
```

Comment:
The usage here with `and` seems to have been a bug.

Reply:
It looks like a bug indeed, but are you sure this is the right fix?

Reply:
It looks correct to me. This part of the frontend could definitely be cleaned up at some point...